includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1034,61 +1034,64 @@ public function get_ending_sales() {
 	 * @return int Matching variation ID or 0.
 	 */
 	public function find_matching_product_variation( $product, $match_attributes = array() ) {
-		$query_args = array(
-			'post_parent' => $product->get_id(),
-			'post_type'   => 'product_variation',
-			'orderby'     => 'menu_order',
-			'order'       => 'ASC',
-			'fields'      => 'ids',
-			'post_status' => 'publish',
-			'numberposts' => 1,
-			'meta_query'  => array(), // phpcs:ignore WordPress.VIP.SlowDBQuery.slow_db_query_meta_query
-		);
-
-		// Allow large queries in case user has many variations or attributes.
-		$GLOBALS['wpdb']->query( 'SET SESSION SQL_BIG_SELECTS=1' );
+		global $wpdb;
 
-		foreach ( $product->get_attributes() as $attribute ) {
-			if ( ! $attribute->get_variation() ) {
-				continue;
+		$matched_variation  = 0;
+		$sorted_meta        = array();
+
+		// Get any variations of the main product
+		$variation_ids = $wpdb->get_results("
+			SELECT ID
+			FROM {$wpdb->prefix}posts
+			WHERE {$wpdb->prefix}posts.post_parent  = " . $product->get_id() . "
+			AND {$wpdb->prefix}posts.post_status	= 'publish'
+			AND {$wpdb->prefix}posts.post_type	  = 'product_variation'
+		", ARRAY_A);
+
+		if ( $variation_ids ) {
+			foreach ( $variation_ids as $ids ) {
+				$variations_string .= $ids['ID'] . ',';
 			}
 
-			$attribute_field_name = 'attribute_' . sanitize_title( $attribute->get_name() );
+			$variations_string = rtrim( $variations_string, ',' );
```

Comment:
Doing what I said about `implode()` and `wp_list_pluck()` you'll not need this anymore.